File: src/test/java/javabot/dao/TestNickServDao.java
package javabot.dao;

import com.google.inject.Inject;
import com.google.inject.Provider;
import javabot.model.NickServInfo;
import org.pircbotx.PircBotX;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public class TestNickServDao extends NickServDao {
    @Inject
    private Provider<PircBotX> ircBot;

    @Override
    public NickServInfo find(final String name) {
        NickServInfo nickServInfo = new NickServInfo(ircBot.get().getUserChannelDao().getUser(name));
        nickServInfo.setRegistered(LocalDateTime.now().minus(30, ChronoUnit.DAYS));
        return nickServInfo;

    }
}

File: src/test/kotlin/javabot/dao/TestNickServDao.kt
package javabot.dao

import dev.morphia.Datastore
import jakarta.inject.Inject
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit
import javabot.model.JavabotUser
import javabot.model.NickServInfo

class TestNickServDao @Inject constructor(ds: Datastore) : NickServDao(ds) {

    override fun find(name: String): NickServInfo {
        val nickServInfo = NickServInfo(JavabotUser(name))
        nickServInfo.registered = LocalDateTime.now().minus(30, ChronoUnit.DAYS)
        return nickServInfo
    }
}
